lib/chef-dk/helpers.rb

```diff
@@ -113,16 +131,37 @@ def git_windows_bin_dir
       @git_windows_bin_dir ||= File.expand_path(File.join(omnibus_root, "embedded", "git", "usr", "bin"))
     end
 
+    def habitat_embedded_bin_dir
+      @habitat_embedded_bin_dir ||= ENV["HAB_WS_EMBEDDED_BIN_DIR"]
+    end
+
+    def habitat_bin_dir
+      @habitat_bin_dir ||= ENV["HAB_WS_BIN_DIR"]
+    end
+
     #
-    # environment vars for omnibus
-    #
+    # provides sane environment variables for running
+    # Workstation and DK tools.
     def omnibus_env
       @omnibus_env ||=
         begin
           user_bin_dir = File.expand_path(File.join(Gem.user_dir, "bin"))
-          path = [ omnibus_bin_dir, user_bin_dir, omnibus_embedded_bin_dir, ENV["PATH"] ]
-          path << git_bin_dir if Dir.exist?(git_bin_dir)
-          path << git_windows_bin_dir if Dir.exist?(git_windows_bin_dir)
+          path = []
+
+          # REVIEW TODO - sanity check
+          #
+          #        cases where the omnibus install exists alongside the hab install
+          #        will get wonky.  Perhaps the omnibus_install? check should be modified to
+          #        check the path of the current ruby executable, looking for platform-appropriate
+          #        "/opt/chef-dk|workstation" in the string?
+          path << omnibus_bin_dir if omnibus_install?
+          path << habitat_bin_dir if habitat_install?
+          path << user_bin_dir
+          path << omnibus_embedded_bin_dir if omnibus_install?
+          path << habitat_embedded_bin_dir if habitat_install?
```

Comment:
if you do combine habitat_bin_dir and habitat_embedded_bin_dir into one directory then it should NOT appear here, and should be BEFORE the user_bin_dir.   this is because the binstubs that do all the pinning MUST appear first -- we don't want a user-installed chef-client to override all the pinning in the binstubs or they'll just wind up going directly into gem dependency hell.  at the same time there should be nothing in the "embedded" directory (mostly ruby itself) which would conflict i think.

Reply:
hmm if there are any "normal" binstubs like pry which aren't appbundled then it should work either way and it'll pick up gems out of the users chef dir via the GEM_PATH anyway, so that also still continues to work...  so yeah, habitat binary dir first, then user_bin_dir then drop embedded.